### PR TITLE
Crates and locker resprite and overhaul

### PR DESCRIPTION
**Pull request by PestoVerde322**
## About The Pull Request

This PR resprites crates and lockers, bringing them to a more modern era, also it overhauls how crates overlays works, implementing proper emissives, alongside a better filter for the Rapid part dispenser and splitting all the storage icons into their own categories.

*Part of the code is inspired by a multitude of PRs over to /tg/*

TODO:
- [x] Mulebots icons
- [x] wrapping/gift icons
## Why It's Good For The Game

Time to modernize the icons and a bit of the code, for everyone's mental sake.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

2FvHJzpZTZ

dreamseeker_pQbFkzxsTH

dreamseeker_TbM0sMFNkh

immagine

</details>

## Changelog
:cl:
imageadd: Redid the icons for the Crates, Closets and the mulebot
code: Overhauled the Crates&Locker code for overlays and emissives
code: Crate climbing times has been re-enabled, now climb time will be different between opened and closed crates!
refactor: Sorted the icons for the storage icons (crates, backpacks, misc storage...)
add: Made the name more intuitive for when looking up the Rapid part dispenser and it's bluespace variant
/:cl:

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -66,31 +81,45 @@
 	return ..()
 
 /obj/structure/closet/update_icon()
```
Comment (Tsar-Salat) [inline code/game/objects/structures/crates_lockers/closets.dm:82]:
```suggestion
/obj/structure/closet/update_icon_state()
```

You split update icon into update_overlays already, might as well do this too.

Recommend changing every call in this file from update_icon to `update_appearance` as well

**Comment by llol111** [comment]
y'know, while it does feel a bit de-souled, suprisingly enough i dont hate these. These are fine.

**Comment by Nakalon** [comment]
This is very nice!!

**Comment by Admiral-Gigaton** [comment]
calm bee reaction in comparison to lynching of the original coder in /TG/

**Review by itsmeow — APPROVED** [review]

**Pull request merged**